backend/views/syson-diagram-statetransition-view/src/main/java/org/eclipse/syson/diagram/statetransition/view/StateTransitionViewDiagramDescriptionProvider.java

```diff
@@ -225,15 +232,21 @@ private NodeTool[] createElementsOfToolSection(IViewDiagramElementFinder cache,
         var nodeTools = new ArrayList<NodeTool>();
 
         elements.forEach(element -> {
-            var nodeDescription = cache.getNodeDescription(this.getDescriptionNameGenerator().getNodeName(element));
-            if (nodeDescription.isPresent()) {
-                nodeTools.add(this.toolDescriptionService.createNodeToolFromDiagramBackground(nodeDescription.get(), element));
-            }
+            cache.getNodeDescription(this.getDescriptionNameGenerator().getNodeName(element))
+                    .ifPresent(nodeDescription -> {
+                        NodeTool nodeTool = null;
+                        if (SysmlPackage.eINSTANCE.getNamespaceImport().equals(element)) {
```

Comment:
Why are you doing this?
It seems strange to this "patch" of this generic code, only for your new tool.
There are other existing tools involving SelectionDialog, and there's no need to do that here for those tools.